### PR TITLE
docs(guides): add the extending world explanation to the new docs

### PR DESCRIPTION
**Pull request by qbzzt**

### CODE REVIEW

File: next-docs/pages/guides/extending-world.mdx [hunk 1]
```diff
@@ -1 +1,484 @@
 # Extending World
+
+On this page you learn how to modify a `World` that is already deployed to the blockchain.
+If you want to learn how to modify a `World` before it is deployed, [see the hello world page](./hello-world).
+
+## The sample program
+
+To learn how to extend a world, we will extend the `Counter` example to allow users to leave a message while incrementing the counter.
```
Comment (alvrs) [inline next-docs/pages/guides/extending-world.mdx:8]:
```suggestion
To learn how to extend a `World`, we will extend the `Counter` example to allow users to leave a message while incrementing the counter.
```

File: next-docs/pages/guides/extending-world.mdx [hunk 2]
```diff
@@ -1 +1,484 @@
...
+
+This requires adding several entities:
```
Comment (alvrs) [inline next-docs/pages/guides/extending-world.mdx:10]:
```suggestion
This requires adding several resources:
```

File: next-docs/pages/guides/extending-world.mdx [hunk 3]
```diff
@@ -1 +1,484 @@
 # Extending World
+
+On this page you learn how to modify a `World` that is already deployed to the blockchain.
+If you want to learn how to modify a `World` before it is deployed, [see the hello world page](./hello-world).
```
Comment (alvrs) [inline next-docs/pages/guides/extending-world.mdx:4]: Should we add one sentence about how everything below assumes there is a default template world deployed already and that we know its world address? (unless i missed that sentence?) For the "how" we can just link to the getting started guide.

Comment (qbzzt) [reply next-docs/pages/guides/extending-world.mdx:4]: It's at line 148, but I also added some language here.

**Review by alvrs — APPROVED** [review]

**Pull request merged**